cmd/internal/kube/daemonset/kube_events_worker.go

```diff
@@ -232,15 +245,25 @@ func (d *Daemonset) inspectPodForEnvVars(pod coreV1.Pod, podArgs *PodArgs) error
 	deployment.SetK8sTraceTags(pod, podArgs.TraceTags)
 
 	podArgs.ContainerUUID = mainContainerUUID
-	err = akid.ParseIDAs(mainContainerConfig.projectID, &podArgs.InsightsProjectID)
+	err = akid.ParseIDAs(mainContainerConfig.requiredContainerConfig.projectID, &podArgs.InsightsProjectID)
 	if err != nil {
 		return errors.Wrap(err, "failed to parse project ID")
 	}
 	podArgs.PodCreds = PodCreds{
-		InsightsAPIKey:      mainContainerConfig.apiKey,
+		InsightsAPIKey:      mainContainerConfig.requiredContainerConfig.apiKey,
 		InsightsEnvironment: d.InsightsEnvironment,
 	}
 
+	// Set ReproMode flag for apidump process
+	reproModeAtDaemonSetLevel := d.InsightsReproModeEnabled
+	reproModeAtPodLevel := !strings.EqualFold(mainContainerConfig.disableReproMode, "true")
+	podArgs.ReproMode = reproModeAtDaemonSetLevel && reproModeAtPodLevel
+	if !reproModeAtDaemonSetLevel {
+		printer.Infof("Repro mode is disabled at the daemonset level for pod: %s\n", pod.Name)
```

Comment:
This seems unnecessary because there's now a startup message warning about repro mode _on_ at the daemonset level.